### ISSUE TITLE
Support creating subtasks

### ISSUE THREAD

**Issue by mickmister** [mattermost/mattermost-plugin-todo#94]
Lots of times, my tasks consist of smaller tasks that all need to be done before the parent task can be considered done. Along with [editing a todo](https://github.com/mattermost/mattermost-plugin-todo/issues/31), being able to attach subtasks would make the workflow of managing your todos a much smoother experience.

Personally, I think this feature would make the plugin exponentially more useful, especially if multiple people can add subtasks to the same task, or share the same task and tackle subtasks together. That's certainly out of scope for this ticket though.
